RocketChat/Rocket.Chat
# fix: Link enclosed in parentheses is not clickable, but link preview is shown

## Proposed changes (including videos or screenshots)

BEFORE THE FIX:
![image](https://github.com/RocketChat/Rocket.Chat/assets/124887084/4c9d7fd1-5b02-450a-924a-79daf0f9d365)

AFTER THE FIX:
![image](https://github.com/RocketChat/Rocket.Chat/assets/124887084/066da0e4-c757-4c00-be8e-c12a0749ac6b)

REASON BEHIND CHANGES:
Earlier when a url was contained in parentheses or apostrophes i wasnt clickable and still showed a preview of the link to fix this i made the changes in the regex so that now to doesnt show the preview if the url isnt passed properly.

## Issue(s)
This should close issues :-
closes [#30710](https://github.com/RocketChat/Rocket.Chat/issues/30710)
## Steps to test or reproduce
Navigate to any channel.
Write a message containing `(https://rocket.chat/)` (make sure to include the parentheses!) and submit it.

## Further comments
